src/authentication/index.ts

```diff
@@ -0,0 +1,122 @@
+/* Facade for keycloak */
+import Keycloak, { KeycloakInitOptions } from 'keycloak-js';
+
+export interface ResourceRole {
+  resource: string;
+  resourceId: string;
+  role: string;
+}
+
+export interface OrganizationRole {
+  orgNr: string;
+  role: string;
+}
+
+export interface AuthConfiguration {
+  oidcIssuer: string;
+  clientId: string;
+  redirectUri: string;
+  logoutRedirectUri: string;
+  silentCheckSsoRedirectUri?: string;
+}
+
+export interface User {
+  username: string;
+  name: string;
+  fdkTerms: string;
+  orgTerms: string[];
+}
+
+export class Auth {
+  private readonly kc: Keycloak;
+
+  constructor(private readonly conf: AuthConfiguration) {
+    const [url, realm] = conf.oidcIssuer.split('/realms/');
+    const kcConfig = { realm, url, clientId: conf.clientId };
+
+    this.conf = conf;
+    this.kc = new Keycloak(kcConfig);
+  }
+
+  init: ({ loginRequired }: { loginRequired: boolean }) => Promise<boolean> = async ({ loginRequired }) => {
+    const keycloakInitOptions: KeycloakInitOptions = {
+      onLoad: 'check-sso',
+      silentCheckSsoRedirectUri: this.conf.silentCheckSsoRedirectUri,
+    };
+    await this.kc.init(keycloakInitOptions).catch();
+    if (loginRequired && !this.isAuthenticated()) {
+      await this.login();
+    }
+    return this.isAuthenticated();
+  };
+
+  login: () => Promise<void> = () => this.kc.login().then().catch();
+
+  logout: () => Promise<void> = () => this.kc.logout({ redirectUri: this.conf.logoutRedirectUri }).then().catch();
+
+  isAuthenticated: () => boolean = () => this.kc.authenticated || false;
+
+  getUser: () => User = () =>
+    this.kc.tokenParsed
+      ? {
+          username: this.kc.tokenParsed.user_name,
+          name: this.kc.tokenParsed.name,
+          fdkTerms: this.kc.tokenParsed.fdk_terms,
+          orgTerms: this.kc.tokenParsed.org_terms.split(','),
+        }
+      : {
+          username: '',
+          name: '',
+          fdkTerms: '',
+          orgTerms: '',
+        };
+
+  getToken: () => Promise<string> = () =>
+    this.kc
+      .updateToken(30)
+      .catch(() => this.logout())
+      .then(() => this.kc.token as string);
+
+  getAuthorizationHeader: () => Promise<string> = async () => `Bearer ${await this.getToken()}`;
+
+  getAuthorities: () => string = () => ((this.kc.tokenParsed && this.kc.tokenParsed.authorities) as string) || '';
```

Comment:
Litt usikker, men er ikke noe slik mer _riktig_ syntax?
```suggestion
  getAuthorities: () => string = () => this.kc.tokenParsed?.authorities ?? '';
```